### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by Rocio01**
This is a pull request for the revision of the project Advanced-Building-Blocks-Enumerables
 In this project, I implemented my own version of Ruby’s enumerable methods.
 - #my_each
 - #my_each_with_index
 - #my_select
 - #my_all?
 - #my_any?
 - #my_none?
 - #my_count
 - #my_map
 - #my_inject
 - Test #my_inject by creating a method called #multiply_els
Thanks for the attention

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
+def my_each(array)
+  w = 0
+  while w < array.size
+    yield(array[w])
+    w += 1
+  end
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_each 

- [ ] Returns an Enumerator if no block is given

     Failure/Error: expect(array.my_each).to be_an(Enumerator)

     *Resulted in NoMethodError*
##

# my_each 

- [ ] Does not mutate the original array

     Failure/Error: array.my_each { |num| num + 1 }

     *Resulted in NoMethodError*


##
 
# my_each 

- [ ] When a block is given when self is an array returns the array itself after calling the given block once for each element in self

     Failure/Error: expect(array.my_each(&block)).to eq(array.each(&block))

     *Resulted in NoMethodError*

##

# my_each 

- [ ] When a block is given when self is an array behaves like #each calls the given block once for each element in self

     Failure/Error: enum.my_each(&block)
     
     *Resulted in NoMethodError*
##

# my_each 

- [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self

     Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))

     *Resulted in NoMethodError*

##

# my_each 
- [ ] When a block is given when self is a range behaves like #each calls the given block once for each element in self

     Failure/Error: enum.my_each(&block)

     *Resulted in NoMethodError*

##

# my_each
 
- [ ] When a block is given when self is a hash behaves like #each calls the given block once for each element in self

     Failure/Error: enum.my_each(&block)

     *Resulted in NoMethodError*

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#  my_each([2,10,12,15]) {|e| puts e + e}
+
+def my_each_with_index(array)
+  w = 0
+  while w < array.size
+    yield(array[w], w)
+    w += 1
+  end
+  array
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_each_with_index 

- [ ] Returns an enumerator if no block is given

     Failure/Error: expect(array.my_each_with_index).to be_an(Enumerator)

     *Resulted in NoMethodError*

# my_each_with_index 

- [ ] Does not mutate the original array

     Failure/Error: array.my_each_with_index { |num| num + 1 }

     *Resulted in NoMethodError*

##

# my_each_with_index 

- [ ] When a block is given when self is an array returns the array itself after calling the given block once for each element in self

      Failure/Error: expect(array.my_each_with_index(&block)).to eq(array.each_with_index(&block))

      *Resulted in NoMethodError*
##

# my_each_with_index 

- [ ] When a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self

      Failure/Error: enum.my_each_with_index(&block)

      *Resulted in NoMethodError*

##

# my_each_with_index 

- [ ] When a block is given when self is a range returns the range itself after calling the given block once for each element in self

      Failure/Error: expect(range.my_each_with_index(&block)).to eq(range.each_with_index(&block))

      *Resulted in NoMethodError*

##

# my_each_with_index 

- [ ] When a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self

      Failure/Error: enum.my_each_with_index(&block)

      *Resulted in NoMethodError*

##

# my_each_with_index 

- [ ] When a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self

      Failure/Error: enum.my_each_with_index(&block)

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+#  my_each_with_index([2,10,12,15]) {|e, index| puts (e + e) if index.odd?}
+
+def my_select(array)
+  second_array = []
+  my_each(array) do |w|
+    second_array.push(w) if yield w
+  end
+  second_array
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_select 

- [ ] Returns an array containing all elements of enum for which the given block returns a true value

      Failure/Error: expect(array.my_select(&block)).to eq(array.select(&block))

      *Resulted in NoMethodError*

##

# my_select 

- [ ] Returns an array containing all elements of enum for which the given block returns a true value::range

      Failure/Error: expect(range.my_select(&block)).to eq(range.select(&block))

      *Resulted in NoMethodError*

##

# my_select 

- [ ] Returns an enumerator if no block is given

      Failure/Error: expect(array.my_select).to be_an(Enumerator)

      *Resulted in NoMethodError*

##

# my_select 

- [ ] Does not mutate the original array

      Failure/Error: array.my_select { |num| num + 1 }

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+# puts my_select([1, 2, 3]){|x| x != 2}
+
+def my_all?(array)
+  my_each(array) do |w|
+    return false if yield(w) == false
+  end
+  true
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_all? 

- [ ] Returns true if the block never returns false or nil

      Failure/Error: expect(array.my_all?(&true_block)).to eq(array.all?(&true_block))

      *Resulted in NoMethodError*

##

# my_all?
 
- [ ] Returns false if the block returns false or nil

      Failure/Error: expect(array.my_all?(&false_block)).to eq(array.all?(&false_block))

      *Resulted in NoMethodError*

##

# my_all?
 
- [ ] Returns false if the block returns false or nil::range

      Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block))

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] Does not mutate the original array
      Failure/Error: array.my_all? { |num| num + 1 }

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When no block or argument is given returns true when none of the collection members is false or nil

      Failure/Error: expect(true_array.my_all?).to be true_array.all?

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When no block or argument is given returns false when one of the collection members is false or nil

      Failure/Error: expect(false_array.my_all?).to be false_array.all?

     *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a class is passed as an argument returns true if all of the collection is a member of such class::Integer

      Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] hen a class is passed as an argument returns true if all of the collection is a member of such class::Numeric

      Failure/Error: expect(numbers.my_all?(Numeric)).to be numbers.all?(Numeric)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a class is passed as an argument returns false if any of the collection is not a member of such class

      Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a Regex is passed as an argument returns true if all of the collection matches the Regex

      Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a Regex is passed as an argument returns false if any of the collection does not match the Regex

      Failure/Error: expect(words.my_all?(/o/)).to be words.all?(/o/)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

      Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

      *Resulted in NoMethodError*

##

# my_all? 

- [ ] When a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

      Failure/Error: expect(array.my_all?(3)).to be array.all?(3)

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+# puts my_all?([1]){|x| x == 1 }
+# puts my_all?(["hello"]){|x| x == 1 }
+
+def my_any?(array)
+  my_each(array) do |w|
+    return true if yield(w) == true
+  end
+  false
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_any? 

- [ ] Returns true if the block ever returns a value other than false or nil

      Failure/Error: expect(array.my_any?(&true_block)).to eq(array.any?(&true_block))

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] Returns false if the block ever returns a value false or nil

      Failure/Error: expect(array.my_any?(&false_block)).to eq(array.any?(&false_block))

      *Resulted in NoMethodError*
##

# my_any? 

- [ ] Returns false if the block ever returns a value false or nil::range

      Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] Does not mutate the original array

      Failure/Error: array.my_any? { |num| num + 1 }

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When no block or argument is given returns true if at least one of the collection is not false or nil

      Failure/Error: expect(true_array.my_any?).to be true_array.any?

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When no block or argument is given returns false if at least one of the collection is not true

      Failure/Error: expect(false_array.my_any?).to be false_array.any?

      *Resulted in NoMethodError*
##

# my_any? 

- [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

      Failure/Error: expect(array.my_any?(Numeric)).to be array.any?(Numeric)

      *Resulted in NoMethodError*

##

# my_any? 
- [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer

      Failure/Error: expect(words.my_any?(Integer)).to be words.any?(Integer)

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When a Regex is passed as an argument returns true if any of the collection matches the Regex

      Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When a Regex is passed as an argument returns false if none of the collection matches the Regex

      Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

      *Resulted in NoMethodError*

##

# my_any? 

- [ ] When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+#  puts my_any?([1,2,3]) { |n| n > 0 }
+# puts my_any?([1,2,3]) { |n| n < 0 }
+
+def my_none?(array)
+  my_each(array) do |w|
+    return false if yield(w) == true
+  end
+  true
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_none? 

- [ ] Returns true if the block never returns true for all elements

      Failure/Error: expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] Returns false if the block ever returns true for all elements

      Failure/Error: expect(array.my_none?(&false_block)).to eq(array.none?(&false_block))

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] Returns false if the block ever returns true for all elements::range

      Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))

      *Resulted in NoMethodError*
# my_none? 

- [ ] Does not mutate the original array

      Failure/Error: array.my_none? { |num| num + 1 }

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When no block or argument is given returns true only if none of the collection members is true

      Failure/Error: expect(false_array.my_none?).to be true

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When no block or argument is given returns false only if one of the collection members is true

      Failure/Error: expect(true_array.my_none?).to be false

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a class is passed as an argument returns true if none of the collection is a member of such class

      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a class is passed as an argument returns true if none of the collection is a member of such class::Numeric

      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a class is passed as an argument returns false if any of the collection is a member of such class

      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a Regex is passed as an argument returns true if none of the collection matches the Regex

      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a Regex is passed as an argument returns false if any of the collection matches the Regex

      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

      *Resulted in NoMethodError*

##

# my_none? 

- [ ] When a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+# puts my_none?([1]){|x| x == 1 }
+# puts my_none?(["hello"]){|x| x == 1 }
+
+def my_count(array)
+  my_select(array) { |w| yield(w) }.size
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_count 

- [ ] Returns the number of items in enum through enumeration

      Failure/Error: expect(array.my_count).to eq array.count

      *Resulted in NoMethodError*

##

# my_count 

- [ ] Returns the number of items in enum through enumeration::range

      Failure/Error: expect(range.my_count).to eq range.count

      *Resulted in NoMethodError*

##

# my_count 

- [ ] Counts the number of items in enum that are equal to an item if an argument is given

      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)

      *Resulted in NoMethodError*

##

# my_count 

- [ ] Counts the number of elements yielding a true value if a block is given

      Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)

      *Resulted in NoMethodError*

##

# my_count 

- [ ] Does not mutate the original array

      Failure/Error: array.my_count { |num| num + 1 }

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+#  puts my_count([1,2,3,4,8]) {|w| w%2 == 0}
+
+def my_map(array, &block)
+  second_array = []
+  my_each(array) do |w|
+    second_array.push(block.call(w))
+  end
+  second_array
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_map 

- [ ] Returns a new array with the results of running a given block once for every element in enum.

      Failure/Error: expect(array.my_map { |num| num < 10 }).to eq(array.map { |num| num < 10 })

      *Resulted in NoMethodError*

##

# my_map 

- [ ] Returns a new array with the results of running a given block once for every element in enum.

      Failure/Error: expect(array.my_map(&block)).to eq array.map(&block)

      *Resulted in NoMethodError*

##

# my_map 

- [ ] Returns a new array with the results of running a given block once for every element in enum::range

      Failure/Error: expect(range.my_map(&block)).to eq range.map(&block)

      *Resulted in NoMethodError*

##

# my_map 

- [ ] Returns an Enumerator if no block is given

      Failure/Error: expect(array.my_map).to be_an(Enumerator)

      *Resulted in NoMethodError*

##

# my_map 

- [ ] Accepts a proc as an argument and returns a new array with the results of running the proc once for every element in enum.

      Failure/Error: expect(array.my_map(&my_proc)).to eq array.map(&my_proc)

      *Resulted in NoMethodError*

##

# my_map 
- [ ] Executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.

      Failure/Error: expect(array.my_map(my_proc) { |num| num < 10 }).to eq array.map(&my_proc)

      *Resulted in NoMethodError*

##

# my_map 

- [ ] Does not mutate the original array

      Failure/Error: array.my_map { |num| num + 1 }

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+# puts my_map([1, 2, 3]){|x| x + 3}
+# puts my_map([1, 2, 3]){|x| x <= 3}
+
+def my_inject(array)
+  my_each(self) do |w|
+    array = yield(array, w)
+  end
+  array
+end
```
Comment (christianotieno) [inline enumerables.rb]: # my_inject 

- [ ] Raises a "LocalJumpError" when no block or argument is given

      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)

        expected LocalJumpError, got *Resulted in NoMethodError*

##

# my_inject 

- [ ] Searches for the longest word in an array of strings

      Failure/Error: expect(words.my_inject(&search)).to eq words.inject(&search)

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] Does not mutate the original array

      Failure/Error: array.my_inject { |num| num + 1 }

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block

      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)

      *Resulted in NoMethodError*

##

# my_inject
 
- [ ] When a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range

      Failure/Error: actual = range.my_inject { |prod, n| prod * n }

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block

      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method

      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range

      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method

      Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)

      *Resulted in NoMethodError*

##

# my_inject 

- [ ] When a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range

      Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)

      *Resulted in NoMethodError*

File: enumerables.rb [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+# puts [1,2,3,4].my_inject(0){|sum, num| sum + num}
+
+def multiply_els(array)
+  array.my_inject(1) { |multiplier, num| multiplier * num }
+end
```
Comment (christianotieno) [inline enumerables.rb]: # multiply_els 

- [ ] Accepts an array as an argument and multiplies all the elements of the array together using #my_inject

      Failure/Error: actual = multiply_els [2, 4, 5]

      *Resulted in NoMethodError*

**Review by christianotieno — CHANGES_REQUESTED** [review]
# **CHANGES REQUIRED** :warning:
Hey @Rocio01  :wave:
-
Good try with the project! :tada: Keep it up.
We, however, do have some few more changes to carry out before we proceed (see review comment).
## Sadly all of your methods are not passing any of my tests. They are all resulting in no method error in all of my tests. Please do a redo of the problems and look down below for inspirations on the same.
Below are the detailed tests that the methods are required to be passing.
##
### NOTE: These are the variables I used in the tests below: 👇 
```
ARRAY_SIZE = 100
LOWEST_VALUE = 0
HIGHEST_VALUE = 9

describe 'enumerables' do
  let(:array) { Array.new(ARRAY_SIZE) { rand(LOWEST_VALUE...HIGHEST_VALUE) } }
  let(:block) { proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } }
  let(:words) { %w[dog door rod blade] }
  let(:range) { Range.new(5, 50) }
  let(:hash) { { a: 1, b: 2, c: 3, d: 4, e: 5 } }
  let(:numbers) { [1, 2i, 3.14] }
  let!(:array_clone) { array.clone }
```
## Tests Done: 79
## Failures: 79
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by Rocio01** [comment]
Thanks for the feedback.

File: enumerables.rb [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+      w = 0
+      while w < size
+        yield(self[w])
+        w += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_each: when a block is given when self is a range behaves like #each calls the given block once for each element in self_**

**_#my_each: when a block is given when self is a hash, behaves like #each calls the given block once for each element in self_**

File: enumerables.rb [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+      w = 0
+      while w < size
+        yield(self[w], w)
+        w += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_each_with_index: when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self_**

**_#my_each_with_index: when a block is given when self is a hash, behaves like #each_with_index calls the given block once for each element in self_**

File: enumerables.rb [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+    if block_given?
+      my_array = []
+      my_each do |w|
+        my_array.push(w) if yield w
+      end
+      my_array
+    else
+      to_enum
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_select returns an array containing all elements of enum for which the given block returns a true value::range_**
**`_block = proc { |num| num < 4 }`_**
**_`(0..9).my_select(&block) # => [0, 1, 2, 3]`_**

File: enumerables.rb [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+      my_each { |w| return false unless w == argument }
+    elsif argument.class == Regexp
+      my_each { |w| return false unless w =~ argument }
+    elsif argument.class == Class
+      my_each { |w| return false unless w.class == argument }
+    else !argument && !block_given?
+         my_each { |w| return false unless w }
+    end
+    true
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_all? returns false if the block returns false or nil::range_**

**_#my_all?(Numeric) when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric_**

File: enumerables.rb [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+      my_each { |w| return true if argument == w }
+    elsif argument.class == Regexp
+      my_each { |w| return true if w =~ argument }
+    elsif argument.class == Class
+      my_each { |w| return true if w.class == argument }
+    else !block_given? && !argument
+         my_each { |w| return true if w }
+    end
+    false
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_any? returns false if the block ever returns a value false or nil::range_**

**_#my_any?(Numeric) when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric_**

File: enumerables.rb [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+      my_each { |w| return false if w == argument }
+    elsif argument.class == Regexp
+      my_each { |w| return false if w =~ argument }
+    elsif argument.class == Class
+      my_each { |w| return false if w.class == argument }
+    else !argument && !block_given?
+         my_each { |w| return false if w }
+    end
+    true
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_none? returns false if the block ever returns true for all elements::range_**

File: enumerables.rb [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+    if block_given?
+      my_select { |w| yield(w) }.size
+    elsif !block_given? && !argument.nil?
+      my_each { |w| x += 1 if argument == w }
+      x
+    else
+      my_each { x += 1 }
+      x
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb:102]: **_#my_count returns the number of items_**
**_`(0..5).my_count # => 6`_**

File: enumerables.rb [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+  #  puts ([1, 2, 4, 23, 34, "w", "w"].my_count("w"))
+
+  def my_map(&block)
+    if block_given?
+      second_array = []
+      my_each { |w| second_array.push(block.call(w)) }
+      second_array
+    else to_enum
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_map returns a new array with the results of running a given block once for every element in enum::range_**

**_#my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum_**

File: enumerables.rb [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+      else
+        sym.nil?
+        if sym.class == Symbol || sym.class == String
+          acc = argument
+          my_each { |w| acc = acc.send(sym, w) }
+          acc
+        end
+      end
+    end
+  end
```
Comment (lekegitrepo) [inline enumerables.rb]: **_#my_inject raises a "LocalJumpError" when no block or argument is given_**

**_#my_inject, when a block is given with an initail value, combines all elements of enum by applying a binary operation, specified by a block_**

**_#my_inject, when a symbol is specified without an initail value, combines each element of the collection by applying the symbol as a named method:range_**

**_#my_inject, when a symbol is specified with an initail value, combines each element of the collection by applying the symbol as a named method::range_**

**_#multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Request changes :recycle:
### 79 Test cases :toolbox: :wavy_dash:  59 Passed :heavy_check_mark: , 20 Failed :x:

- **_Remove all the commented codes_**

**_Fix these required changes and submit them for review._**

**_Check my comments below for required changes :point_down: _**

**Review by rOluochKe — APPROVED** [review]
Hello @Rocio01 
### Well done, you got this right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**